Check the version number in the WebbPSF data package on startup

josePhoenix opened the pull request:
Adds a constant in `webbpsf.__init__` called `DATA_VERSION_MIN` to check against, and some logic that runs at import.

mperrin commented on webbpsf/utils.py:
```diff
@@ -107,7 +107,7 @@ def setup_logging(level='INFO',  filename=None):
 
 MISSING_WEBBPSF_DATA_MESSAGE = """
- *********  ERROR  ******  ERROR  ******  ERROR  ******  ERROR  *************
+ ***********  ERROR  ******  ERROR  ******  ERROR  ******  ERROR  ***********
```
I appreciate your attention to detail in symmetry of asterisks. :-)

josePhoenix replied:
Thanks :smile:

mperrin commented on webbpsf/utils.py:
```diff
@@ -147,78 +147,30 @@ def get_webbpsf_data_path():
+        version_tuple = tuple(map(int, parts))
+    except (IOError, ValueError):
+        raise EnvironmentError("Couldn't read the version number from {}. (Do you need to "
+                               "update the WebbPSF data?)".format(version_file_path))
```
Let's add the "See [URL] for a link to the latest version" text to this error message too, for consistency and maximum helpfulness.

josePhoenix replied:
Good call, done!

mperrin commented:
Looks good. Although it breaks the Travis build entirely so my next order of business is updating the minimal-data.tar.gz file.

mperrin commented:
Let me know when you've gotten a chance to update the data files on Central Store accordingly and I will update the tar files.

josePhoenix commented:
Made the following changes to `/itar/jwst/tel/share/webbpsf/webbpsf-data-source`:

* Added `BUNIT` keyword with value `micron` to every OPD map
* Added the 2048px masked and unmasked WFIRST pupils
* Moved the `WFIRSTImager` data folder to `WFI`
* Updated `zernikes.csv` in there
* Deleted `WFIRSTSpectrograph` folder since it didn't have anything in it
* Added `upscaled_HST_OPD.fits` (with `BUNIT` of `meter`)
* Added `add_OPD_header_keyword_BUNIT.py` in case we want to give it to people who would rather not download all the OPD files again

mperrin commented:
I have updated the minimal-data.tar.gz file and kicked Travis to try again.  If that works I will put up the new public data tar file so it's ready later whenever the next code release is, or for anyone using `master` directly from Github, and then merge this.

mperrin commented:
Wait, no, even after updating the data files these changes break the code. See 
https://travis-ci.org/mperrin/webbpsf/jobs/76130411

The issue is all the usages of `get_webbpsf_data_path()` by the instrument classes in `__init__` now fail since the function call is expecting the minimum version tuple.  Suggestion: make `data_version_min` optional, and if it's `None` then just don't perform that check.

josePhoenix commented:
Ah, whoops, serves me right for trying to rush to lunch.

josePhoenix commented:
Merging so we can see how this behaves in JupyterHub cc @cslocum
